maliciousscan_test.go

```diff
@@ -0,0 +1,168 @@
+package main
+
+import (
+	"path/filepath"
+	"strconv"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/jfrog/jfrog-cli-core/v2/common/format"
+	securityTests "github.com/jfrog/jfrog-cli-security/tests"
+	securityTestUtils "github.com/jfrog/jfrog-cli-security/tests/utils"
+	securityIntegrationTestUtils "github.com/jfrog/jfrog-cli-security/tests/utils/integration"
+	"github.com/jfrog/jfrog-cli-security/tests/validations"
+)
+
+// test malicious-scan command parameters
+type maliciousScanCommandTestParams struct {
+	// Will be combined with "," if provided and be used as --working-dirs flag value
+	WorkingDirsToScan []string
+	// --format flag value if provided
+	Format format.OutputFormat
+	// --threads flag value if provided
+	Threads int
+	// --min-severity flag value if provided
+	MinSeverity string
+}
+
+func getMaliciousScanCmdArgs(params maliciousScanCommandTestParams) (args []string) {
+	args = []string{"malicious-scan"}
+	if len(params.WorkingDirsToScan) > 0 {
+		workingDirs := ""
+		for i, dir := range params.WorkingDirsToScan {
+			if i > 0 {
+				workingDirs += ","
+			}
+			workingDirs += dir
+		}
+		args = append(args, "--working-dirs="+workingDirs)
+	}
+	if params.Format != "" {
+		args = append(args, "--format="+string(params.Format))
+	}
+	if params.Threads > 0 {
+		args = append(args, "--threads="+strconv.Itoa(params.Threads))
+	}
+	if params.MinSeverity != "" {
+		args = append(args, "--min-severity="+params.MinSeverity)
+	}
+	return args
+}
+
+func testMaliciousScan(t *testing.T, params maliciousScanCommandTestParams, errorExpected bool) string {
+	output, err := runMaliciousScan(t, params)
+	if errorExpected {
+		assert.Error(t, err)
+	} else {
+		assert.NoError(t, err)
+	}
+	return output
+}
+
+func runMaliciousScan(t *testing.T, params maliciousScanCommandTestParams) (string, error) {
+	cleanUp := securityIntegrationTestUtils.UseTestHomeWithDefaultXrayConfig(t)
+	defer cleanUp()
+	return securityTests.PlatformCli.RunCliCmdWithOutputs(t, getMaliciousScanCmdArgs(params)...)
+}
+
+func TestMaliciousScanWithMaliciousFile(t *testing.T) {
```

Comment:
You can combine `TestMaliciousScanWithMaliciousFile` and `TestMaliciousScanNoMaliciousFile`.
Just add another testCase attribute `expected issues` (and if expected issues == 0 don't forget to add `validationsParams.validationsParams.ExactResultsMatch = true`)